packages/main/src/plugin/configuration-registry.ts

```diff
@@ -76,19 +77,25 @@ export class ConfigurationRegistry implements IConfigurationRegistry {
     return path.resolve(this.directories.getConfigurationDirectory(), 'settings.json');
   }
 
-  public init(): NotificationCardOptions[] {
+  public async init(): Promise<NotificationCardOptions[]> {
     const notifications: NotificationCardOptions[] = [];
 
     const settingsFile = this.getSettingsFile();
     const parentDirectory = path.dirname(settingsFile);
-    if (!fs.existsSync(parentDirectory)) {
-      fs.mkdirSync(parentDirectory, { recursive: true });
+
+    try {
+      await fsPromises.access(parentDirectory);
+    } catch {
+      await fsPromises.mkdir(parentDirectory, { recursive: true });
     }
```

Comment:
mkdir with recursive option internally check for existence before doing the call, let's avoid doing twice the work and directly mkdir


```suggestion
    await fsPromises.mkdir(parentDirectory, { recursive: true });
```

Reply:
Hahaha hoping that each subsequent call is only called once! But we should be checking if the directory is available before creating it, no?

This follows the same logic as the previous method where we make sure the directory exists before the mkdir function.

Reply:
Keep in mind this looks different since we can't use `existsSync` anymore / prefer using promises.

Reply:
> Hahaha hoping that each subsequent call is only called once! But we should be checking if the directory is available before creating it, no?

I guess that's fine 👍